### PR TITLE
Add history to expression textarea

### PR DESCRIPTION
**Pull request by wind0r**
As suggested in Issue #1103 from @fabxc 

Keys are shift and j/k for history up and down.
Maybe some other keys should be choose but i did not find a better combination

---

This change is [Reviewable](https://reviewable.io/reviews/prometheus/prometheus/1188)

### CODE REVIEW

File: web/blob/static/js/graph.js [hunk 1]
```diff
@@ -61,6 +63,18 @@ Prometheus.Graph.prototype.initialize = function() {
       e.preventDefault();
     }
 
+    if (e.which == 74 && e.shiftKey && self.expr.curElement > 0) {
```
Comment (grobie) [inline web/blob/static/js/graph.js]: Can you extract the magic number into a "constant" please (however to solve that best in javascript). A clear name describing the key helps other readers to understand what's going on.

**Comment by fabxc** [comment]
Thanks a lot! That's a good idea.

Unfortunately I found another bug affecting `CTRL+Z` in the original code. With all those weirdnesses going on I have a hard time testing this as I run into bugs along the way.

It would probably be great to have the issues described in #1185 figured out before adding a new feature.

**Comment by grobie** [comment]
That's a great start! Thank you.

My main use case is that I run some ad-hoc query at one day which might took me some minutes to get right and became a bit more complicated. Then, the next day, I need to run that again. Chances are high that I closed the tab in the meantime.

Would it be possible to store the history in the browser's local storage? Maybe by adding to elements to the top of the page `[x] Store history of last 100 expressions in local storage.` and a button `[Clear history now]`.

**Comment by Wind0r** [comment]
Currently each expression textarea get its own history.
Maybe while appending to that history we also append the input value to a local storage.
So if the Site is reloaded or a textarea is added the default history will be filled with the one from the  local storage

**Comment by grobie** [comment]
@Wind0r Sorry for the late feedback. Can you rebase against current master?

**Comment by brian-brazil** [comment]
Review is unfortunately quite stale by now.

**Pull request closed without merging**